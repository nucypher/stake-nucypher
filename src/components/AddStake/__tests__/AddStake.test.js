import React from 'react';
import AddStake from '../AddStake';
import { render, fireEvent, wait } from '@testing-library/react'

describe('<AddStake />', () => {

  it('should render correctly', () => {
    const { asFragment } = render(<AddStake />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger onAddStake on adding stake', async () => {
    const onAddStake = jest.fn(() => {});
    const { getByLabelText, getByText } = render(<AddStake onAddStake={onAddStake} />);

    await wait(() => {
      fireEvent.change(getByLabelText('Stake value'), { target: { value: '15000' } });
      fireEvent.change(getByLabelText('Stake duration'), { target: { value: '365' } });
      fireEvent.click(getByText('Stake'));
    });

    expect(onAddStake).toBeCalledWith(expect.objectContaining({ stakeValue: 15000, stakeDuration: 365, infiniteApproval: false }));
  });

  it('should validate input from user correctly', async () => {
    const { getByLabelText, getByText, asFragment } = render(<AddStake />);

    await wait(() => {
      fireEvent.change(getByLabelText('Stake value'), { target: { value: '1' } });
      fireEvent.change(getByLabelText('Stake duration'), { target: { value: '1' } });
      fireEvent.click(getByText('Stake'));
    });

    expect(asFragment()).toMatchSnapshot();

    await wait(() => {
      fireEvent.change(getByLabelText('Stake value'), { target: { value: '' } });
      fireEvent.change(getByLabelText('Stake duration'), { target: { value: '' } });
      fireEvent.click(getByText('Stake'));
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
