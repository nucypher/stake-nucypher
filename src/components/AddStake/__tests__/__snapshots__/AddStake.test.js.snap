// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddStake /> should render correctly 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-4"
        for="stake-value"
      >
        Stake value
      </label>
      <div
        class="col-sm-8"
      >
        <input
          class="form-control"
          id="stake-value"
          name="stakeValue"
          type="number"
          value=""
        />
        <div
          class="feedback-placeholder"
        >
          <div
            class="invalid-feedback"
            style="display: none;"
          />
        </div>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-4"
        for="stakeDuration"
      >
        Stake duration
      </label>
      <div
        class="col-sm-8"
      >
        <input
          class="form-control"
          id="stakeDuration"
          name="stakeDuration"
          type="number"
          value=""
        />
        <div
          class="feedback-placeholder"
        >
          <div
            class="invalid-feedback"
            style="display: none;"
          />
        </div>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="col-sm-8 offset-sm-4"
      >
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            name="infiniteApproval"
            type="checkbox"
            value="false"
          />
          <label
            class="form-check-label"
            title=""
          >
            Infinite approval
          </label>
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        type="submit"
      >
        Stake
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<AddStake /> should validate input from user correctly 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-4"
        for="stake-value"
      >
        Stake value
      </label>
      <div
        class="col-sm-8"
      >
        <input
          class="form-control"
          id="stake-value"
          name="stakeValue"
          type="number"
          value=""
        />
        <div
          class="feedback-placeholder"
        >
          <div
            class="invalid-feedback"
            style="display: inline;"
          >
            Minimum stake is 15000 NU
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-4"
        for="stakeDuration"
      >
        Stake duration
      </label>
      <div
        class="col-sm-8"
      >
        <input
          class="form-control"
          id="stakeDuration"
          name="stakeDuration"
          type="number"
          value=""
        />
        <div
          class="feedback-placeholder"
        >
          <div
            class="invalid-feedback"
            style="display: inline;"
          >
            Minimum duration is 30 periods
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="col-sm-8 offset-sm-4"
      >
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            name="infiniteApproval"
            type="checkbox"
            value="false"
          />
          <label
            class="form-check-label"
            title=""
          >
            Infinite approval
          </label>
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        disabled=""
        type="submit"
      >
        Stake
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<AddStake /> should validate input from user correctly 2`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-4"
        for="stake-value"
      >
        Stake value
      </label>
      <div
        class="col-sm-8"
      >
        <input
          class="form-control"
          id="stake-value"
          name="stakeValue"
          type="number"
          value=""
        />
        <div
          class="feedback-placeholder"
        >
          <div
            class="invalid-feedback"
            style="display: inline;"
          >
            Field is required.
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-4"
        for="stakeDuration"
      >
        Stake duration
      </label>
      <div
        class="col-sm-8"
      >
        <input
          class="form-control"
          id="stakeDuration"
          name="stakeDuration"
          type="number"
          value=""
        />
        <div
          class="feedback-placeholder"
        >
          <div
            class="invalid-feedback"
            style="display: inline;"
          >
            Field is required.
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="col-sm-8 offset-sm-4"
      >
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            name="infiniteApproval"
            type="checkbox"
            value="false"
          />
          <label
            class="form-check-label"
            title=""
          >
            Infinite approval
          </label>
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        disabled=""
        type="submit"
      >
        Stake
      </button>
    </div>
  </form>
</DocumentFragment>
`;
